src/helpers.js

```diff
@@ -2,40 +2,73 @@ const fs = require('fs');
 const path = require('path');
 
 /**
- * 
+ *
  * @typedef {Object} AbellConfigs - Configurations from abell.config.js
  * @property {String} contentPath - Path to content directory (default 'content')
  * @property {String} sourcePath - Path to source directory (default 'src')
  * @property {String} destinationPath
  *  - Path to output destination (default 'dist', changes to 'debug' during dev-server)
  * @property {Object} globalMeta - Meta variables that are accessible globally in .abell files
- * 
+ *
  */
 
+const relativeJoinedPath = (pathString) =>
+  path.join(process.cwd(), ...pathString.split('/'));
 
-const relativeJoinedPath = pathString => path.join(process.cwd(), ...pathString.split('/'));
-
-const getDirectories = source => {
-  return fs.readdirSync(source)
-    .filter(dirent => {
-      return fs.lstatSync(path.join(source, dirent)).isDirectory();
-    });
+const getDirectories = (source) => {
+  return fs.readdirSync(source).filter((dirent) => {
+    return fs.lstatSync(path.join(source, dirent)).isDirectory();
+  });
 };
- 
+
 const rmdirRecursiveSync = function(pathToRemove) {
   if (fs.existsSync(pathToRemove)) {
     fs.readdirSync(pathToRemove).forEach((file, index) => {
       const curPath = path.join(pathToRemove, file);
-      if (fs.lstatSync(curPath).isDirectory()) { // recurse
+      if (fs.lstatSync(curPath).isDirectory()) {
+        // recurse
         rmdirRecursiveSync(curPath);
-      } else { // delete file
+      } else {
+        // delete file
         fs.unlinkSync(curPath);
       }
     });
     fs.rmdirSync(pathToRemove);
   }
 };
 
+const recFindByExt = (base, ext, inputFiles, inputResult) => {
+  const files = inputFiles || fs.readdirSync(base);
+  let result = inputResult || [];
+
+  files.map(function(file) {
+    const newbase = path.join(base, file);
+    if (fs.statSync(newbase).isDirectory()) {
+      result = recFindByExt(newbase, ext, fs.readdirSync(newbase), result);
+    } else {
+      if (file.substr(-1 * (ext.length + 1)) == '.' + ext) {
+        result.push(newbase);
+      }
+    }
+  });
+  return result;
+};
+
+// Returns all .abell files in src folder except for [$slug]
+const getAbellFiles = (sourcePath, extension) => {
+  const absolutePaths = recFindByExt(sourcePath, extension.split('.')[1]);
```

Comment:
The right way to find an extension would be using `path.extname(sourcePath)` in nodejs. This code can break when you have a file something like `abell.config.js` where index 1 would have 'config' instead of 'js'

Reply:
Hey, I agree with `path.extname` there. But I would still have to split on `.` after getting the extension, because the function I added expects an extension without a `.`

Reply:
But actually, this `extension` is passed from `programInfo`, so wouldn't we explicitly want to get those files if we're passing `.abell.config.js` as an extension ?

Reply:
Nvm, I refactored the code so extension is passed with a `.` in it, and the function handles the `.` correctly. So no splitting or `path.extname` needed.

Reply:
> Nvm, I refactored the code so extension is passed with a `.` in it, and the function handles the `.` correctly. So no splitting or `path.extname` needed.

This sounds a lot better! thanks

Reply:
This is refactored